HOC/MotionWrapper.tsx

```diff
@@ -0,0 +1,49 @@
+'use client';
+
+import { MotionWrapperType } from '@/types';
+import { fadeIn } from '@/utils/motion';
+import { motion } from 'framer-motion';
+
+const MotionWrapper = ({
+  children,
+  direction,
+  type,
+  delay,
+  duration,
+  className,
+  once = true,
+  tagType = 'div',
+  translateX = '0',
+  translateY = ' 0',
+}: MotionWrapperType) => {
+  let Tag;
+  switch (tagType) {
+    case 'h1':
+      Tag = motion.h1;
+      break;
+    case 'h2':
+      Tag = motion.h2;
+      break;
+    case 'h3':
+      Tag = motion.h3;
+      break;
+    case 'h4':
+    default:
+      Tag = motion.div;
+      break;
```

Comment:
This was a fantastic suggestion by Mentor Brandon. Another approach you can consider is that instead of using a switch statement to determine the tag type, you could use an object that maps tag types to their corresponding motion components. This approach is more concise and easier to maintain.

You could import tagTypeMapping as below and then refer to the Tag.
```
const tagTypeMapping = {
  h1: motion.h1,
  h2: motion.h2,
  h3: motion.h3,
  h4: motion.h4,
};

const Tag = tagTypeMapping[tagType] || motion.div;
```

Reply:
Much better

Reply:
@brandonetter 

Very neat solution.